numpy/ctypeslib.py

```diff
@@ -401,8 +401,9 @@ def prep_pointer(pointer_obj, shape):
         attach an __array_interface__ property to it if it does not
         yet have one.
         """
-        try: pointer_obj.__array_interface__
+        try: pointer_obj.__array_interface__['shape'] = shape
         except AttributeError: pass
+        except TypeError: return
```

Comment:
The `AttributeError` identifies the absent of `__array_interface__`. The `TypeError` states that `__array_interface__` is not a dictionary and `__array_interface__['shape'] = shape` failed. Returning from this will delegate the error handling to `np.array`.

Personally I think, that having another error handling block to handle the second property, is a redundancy that should be skipped. But maybe it makes sense to document the meaning of each exception.

        try: pointer_obj.__array_interface__['shape'] = shape
        # __array_interface__ needs to be setup
        except AttributeError: pass
        # np.array should handle non-dict __array_interface__
        except TypeError: return
        # __array_interface__ is setup and shape was updated
        else: return